allennlp/training/callbacks/should_validate.py

```diff
@@ -0,0 +1,41 @@
+from typing import Dict, Any, TYPE_CHECKING, Optional
+
+from allennlp.training.callbacks.callback import TrainerCallback
+
+if TYPE_CHECKING:
+    from allennlp.training.gradient_descent_trainer import GradientDescentTrainer
+
+
+@TrainerCallback.register("should_validate_callback")
+class ShouldValidateCallback(TrainerCallback):
+    """
+    A callback that you can pass to the `GradientDescentTrainer` to change the frequency of
+    validation during training. If `start_validation` is not `None`, validation will not occur until
```

Comment:
I think `validation_start` sounds a little better than `start_validation`

Reply:
Changed! Yeah, I was struggling with the naming, if you have any other name change suggestions let me know!